nav2_bt_navigator/include/nav2_bt_navigator/ros_topic_logger.hpp

```diff
@@ -38,7 +38,8 @@ class RosTopicLogger : public BT::StatusChangeLogger
   void flush() override;
 
 protected:
-  rclcpp::Node::SharedPtr ros_node_;
+  rclcpp::Clock::SharedPtr clock_;
+  rclcpp::Logger logger_{rclcpp::get_logger("bt_navigator")};
```

Comment:
I think the idea here was to default the logger object to a reasonable name in case the logger object isn't overridden by a user since a lot of the places that use the logger object are plugin interfaces that could be further implemented by an external user.

Reply:
I would agree with @mikeferguson, if we were going to set them, then I think this is the way to do it, but I think we should ensure that they're always set properly from the `node`. For the case of new plugins, they'll be given a new `WeakPtr` for them to own and they'll need to store their own `logger_` to use. For everything in the codebase itself, we should ensure that its always properly set so that it respects remaps.

Reply:
This is a nice safety mechanism for the specific nodes where that mistake may be possible (costmap_2d comes to mind). I'm not sure that is possible in other places like this instance. We "own" BT navigator's node and logger object, there's no plugin coming along to not properly initialize it